### PR TITLE
Refactor docs to live on their own

### PR DESCRIPTION
**Pull request by btobin**
- Removes marketing pages
- Removes all unused marketing assets
- Moves all docs pages to src/
- Updates all references to /docs/
- Updates all marketing pages for new url
- Adds config file for staging urls and testing across 2 sites, including updated docker for passing configs to build
- Adds extends main nav with bootstrap dropdown menus
- Adds staging action for testing
- Adds new Firebase default project 
- Removes unused NPM package
- Upgrade Node to v17

### CODE REVIEW

File: src/cookbook/effects/download-button.md [hunk 1]
```diff
@@ -3,10 +3,10 @@ title: Create a download button
 description: How to implement a download button.
 prev:
   title: Work with tabs
-  path:  /docs/cookbook/design/tabs
+  path:  /cookbook/design/tabs
```
Comment (AlexV525) [inline src/cookbook/effects/download-button.md:6]:
```suggestion
  path: /cookbook/design/tabs
```

File: src/cookbook/effects/download-button.md [hunk 2]
```diff
@@ -3,10 +3,10 @@ title: Create a download button
...
 next:
   title: Create a nested navigation flow
-  path:  /docs/cookbook/effects/nested-nav
+  path:  /cookbook/effects/nested-nav
```
Comment (AlexV525) [inline src/cookbook/effects/download-button.md:9]:
```suggestion
  path: /cookbook/effects/nested-nav
```

File: src/cookbook/effects/expandable-fab.md [hunk 3]
```diff
@@ -3,10 +3,10 @@ title: Create an expandable FAB
 description: How to implement a FAB that expands to multiple buttons when tapped.
 prev:
   title: Create a typing indicator
-  path:  /docs/cookbook/effects/typing-indicator
+  path:  /cookbook/effects/typing-indicator
```
Comment (AlexV525) [inline src/cookbook/effects/expandable-fab.md:6]:
```suggestion
  path: /cookbook/effects/typing-indicator
```

File: src/cookbook/effects/expandable-fab.md [hunk 4]
```diff
@@ -3,10 +3,10 @@ title: Create an expandable FAB
...
 next:
   title: Create gradient chat bubbles
-  path:  /docs/cookbook/effects/gradient-bubbles
+  path:  /cookbook/effects/gradient-bubbles
```
Comment (AlexV525) [inline src/cookbook/effects/expandable-fab.md:9]:
```suggestion
  path: /cookbook/effects/gradient-bubbles
```

File: src/cookbook/effects/gradient-bubbles.md [hunk 5]
```diff
@@ -3,10 +3,10 @@ title: Create gradient chat bubbles
 description: How to implement gradient chat bubbles.
 prev:
   title: Create an expandable FAB
-  path:  /docs/cookbook/effects/expandable-fab
+  path:  /cookbook/effects/expandable-fab
```
Comment (AlexV525) [inline src/cookbook/effects/gradient-bubbles.md:6]:
```suggestion
  path: /cookbook/effects/expandable-fab
```

File: src/cookbook/effects/gradient-bubbles.md [hunk 6]
```diff
@@ -3,10 +3,10 @@ title: Create gradient chat bubbles
...
 next:
   title: Drag a UI element
-  path:  /docs/cookbook/effects/drag-a-widget
+  path:  /cookbook/effects/drag-a-widget
```
Comment (AlexV525) [inline src/cookbook/effects/gradient-bubbles.md:9]:
```suggestion
  path: /cookbook/effects/drag-a-widget
```

File: src/cookbook/effects/nested-nav.md [hunk 7]
```diff
@@ -3,10 +3,10 @@ title: Create a nested navigation flow
 description: How to implement a flow with nested navigation.
 prev:
   title: Create a download button
-  path:  /docs/cookbook/effects/download-button
+  path:  /cookbook/effects/download-button
```
Comment (AlexV525) [inline src/cookbook/effects/nested-nav.md:6]:
```suggestion
  path: /cookbook/effects/download-button
```

File: src/cookbook/effects/nested-nav.md [hunk 8]
```diff
@@ -3,10 +3,10 @@ title: Create a nested navigation flow
...
 next:
   title: Create a photo filter carousel
-  path:  /docs/cookbook/effects/photo-filter-carousel
+  path:  /cookbook/effects/photo-filter-carousel
```
Comment (AlexV525) [inline src/cookbook/effects/nested-nav.md:9]:
```suggestion
  path: /cookbook/effects/photo-filter-carousel
```

File: src/cookbook/effects/parallax-scrolling.md [hunk 9]
```diff
@@ -3,10 +3,10 @@ title: Create a scrolling parallax effect
 description: How to implement a scrolling parallax effect.
 prev:
   title: Create a photo filter carousel
-  path: /docs/cookbook/effects/photo-filter-carousel
+  path: /cookbook/effects/photo-filter-carousel
 next:
   title: Create a shimmer loading effect
-  path:  /docs/cookbook/effects/shimmer-loading
+  path:  /cookbook/effects/shimmer-loading
```
Comment (AlexV525) [inline src/cookbook/effects/parallax-scrolling.md:9]:
```suggestion
  path: /cookbook/effects/shimmer-loading
```

File: src/cookbook/effects/photo-filter-carousel.md [hunk 10]
```diff
@@ -3,10 +3,10 @@ title: Create a photo filter carousel
 description: How to implement a photo filter carousel.
 prev:
   title: Create a nested navigation flow
-  path:  /docs/cookbook/effects/nested-nav
+  path:  /cookbook/effects/nested-nav
```
Comment (AlexV525) [inline src/cookbook/effects/photo-filter-carousel.md:6]:
```suggestion
  path: /cookbook/effects/nested-nav
```

File: src/cookbook/effects/photo-filter-carousel.md [hunk 11]
```diff
@@ -3,10 +3,10 @@ title: Create a photo filter carousel
...
 next:
   title: Create a scrolling parallax effect
-  path:  /docs/cookbook/effects/parallax-scrolling
+  path:  /cookbook/effects/parallax-scrolling
```
Comment (AlexV525) [inline src/cookbook/effects/photo-filter-carousel.md:9]:
```suggestion
  path: /cookbook/effects/parallax-scrolling
```

File: src/cookbook/effects/shimmer-loading.md [hunk 12]
```diff
@@ -3,10 +3,10 @@ title: Create a shimmer loading effect
 description: How to implement a shimmer loading effect.
 prev:
   title: Create a scrolling parallax effect
-  path:  /docs/cookbook/effects/parallax-scrolling
+  path:  /cookbook/effects/parallax-scrolling
```
Comment (AlexV525) [inline src/cookbook/effects/shimmer-loading.md:6]:
```suggestion
  path: /cookbook/effects/parallax-scrolling
```

File: src/cookbook/effects/shimmer-loading.md [hunk 13]
```diff
@@ -3,10 +3,10 @@ title: Create a shimmer loading effect
...
 next:
   title: Create a staggered menu animation
-  path:  /docs/cookbook/effects/staggered-menu-animation
+  path:  /cookbook/effects/staggered-menu-animation
```
Comment (AlexV525) [inline src/cookbook/effects/shimmer-loading.md:9]:
```suggestion
  path: /cookbook/effects/staggered-menu-animation
```

File: src/cookbook/effects/staggered-menu-animation.md [hunk 14]
```diff
@@ -3,10 +3,10 @@ title: Create a staggered menu animation
 description: How to implement a staggered menu animation.
 prev:
   title: Create a shimmer loading effect
-  path:  /docs/cookbook/effects/shimmer-loading
+  path:  /cookbook/effects/shimmer-loading
```
Comment (AlexV525) [inline src/cookbook/effects/staggered-menu-animation.md:6]:
```suggestion
  path: /cookbook/effects/shimmer-loading
```

File: src/cookbook/effects/staggered-menu-animation.md [hunk 15]
```diff
@@ -3,10 +3,10 @@ title: Create a staggered menu animation
...
 next:
   title: Create a typing indicator
-  path:  /docs/cookbook/effects/typing-indicator
+  path:  /cookbook/effects/typing-indicator
```
Comment (AlexV525) [inline src/cookbook/effects/staggered-menu-animation.md:9]:
```suggestion
  path: /cookbook/effects/typing-indicator
```

File: src/cookbook/effects/typing-indicator.md [hunk 16]
```diff
@@ -3,10 +3,10 @@ title: Create a typing indicator
 description: How to implement a typing indicator.
 prev:
   title: Create a staggered menu animation
-  path:  /docs/cookbook/effects/staggered-menu-animation
+  path:  /cookbook/effects/staggered-menu-animation
```
Comment (AlexV525) [inline src/cookbook/effects/typing-indicator.md:6]:
```suggestion
  path: /cookbook/effects/staggered-menu-animation
```

File: src/cookbook/effects/typing-indicator.md [hunk 17]
```diff
@@ -3,10 +3,10 @@ title: Create a typing indicator
...
 next:
   title: Create an expandable FAB
-  path:  /docs/cookbook/effects/expandable-fab
+  path:  /cookbook/effects/expandable-fab
```
Comment (AlexV525) [inline src/cookbook/effects/typing-indicator.md:9]:
```suggestion
  path: /cookbook/effects/expandable-fab
```

**Review by AlexV525 — COMMENTED** [review]
I can't get a smooth experience in this PR :(

**Comment by AlexV525** [comment]
The env in the link validate check needs to be updated too.

**Comment by btobin** [comment]
Hi @AlexV525. I cleaned up the extra space after `path: `, good catch. But the `/docs` you have found might be stale? 

Can you point me to "env in the link validate check". I've commented out link checking in `test.sh` since it's no longer working in the build:

https://github.com/flutter/website/pull/6454/files#diff-3c59f6df95cb9632a5c6331e1904344d39a0030ec44b9ea35a693e7ab41463bc

**Comment by AlexV525** [comment]
> But the `/docs` you have found might be stale?

That might be true. Seems all good now.

> Can you point me to "env in the link validate check".

I'm referencing the "HTML link references validate" check action with GitHub, like this: https://github.com/flutter/website/runs/4191931096?check_suite_focus=true.
Previously we set the node version to 15 and it should be passed if it gets updated too.

**Comment by btobin** [comment]
Thanks @AlexV525. I went ahead and removed Node from that workflow.

**Review by RedBrogdon — APPROVED** [review]

File: src/development/tools/sdk/release-notes/release-notes-1.12.13.md [hunk 18]
```diff
@@ -2379,7 +2379,7 @@ We added the ability to get id in the callback in the Android Alarm Manager plug
 
 #### Plugin: Android Intent
 
-We made several improvements in the Android Intent plugin including adding the ability to pass intent flags (contributed by a community member!), and upgrading it to the [new plugin API](https://github.com/flutter/website/blob/staging-add-to-app-do-not-delete-until-2020/src/docs/development/packages-and-plugins/plugin-api-migration.md).
+We made several improvements in the Android Intent plugin including adding the ability to pass intent flags (contributed by a community member!), and upgrading it to the [new plugin API](https://github.com/flutter/website/blob/staging-add-to-app-do-not-delete-until-2020/sr/development/packages-and-plugins/plugin-api-migration.md).
```
Comment (khanhnwin) [inline src/development/tools/sdk/release-notes/release-notes-1.12.13.md:2382]: Find and replace mistake? Looks like it took `src/docs` replaced `/docs` and ended up with `sr`

Comment (btobin) [reply src/development/tools/sdk/release-notes/release-notes-1.12.13.md]: Yes! Just pushed those changes up.

File: firebase.json [hunk 19]
```diff
@@ -10,112 +10,162 @@
...
+    "rewrites": [],
     "redirects": [
       { "regex": "^/(flutter(/.*)?)", "destination": "https://api.flutter.dev/:1", "type": 301 },
       { "regex": "^/(objcdoc(/.*)?)", "destination": "https://api.flutter.dev/:1", "type": 301 },
       { "regex": "^/(javadoc(/.*)?)", "destination": "https://api.flutter.dev/:1", "type": 301 },
       { "regex": "(?P<basename>.*)\\.html$", "destination": ":basename", "type": 301 },
       { "regex": "(?P<basename>.*)\\.$", "destination": ":basename", "type": 301 },
+
+      { "source": "/community", "destination": "https://flutter.dev/community", "type": 301 },
+      { "source": "/showcase", "destination": "https://flutter.dev/showcase", "type": 301 },
```
Comment (johnpryan) [inline firebase.json:22]: This doesn't seem right - what will happen when this gets deployed to `flutter.dev`?

File: firebase.json [hunk 20]
```diff
@@ -10,112 +10,162 @@
...
+      { "source": "/subscribe", "destination": "https://flutter.dev", "type": 301 },
```
Comment (johnpryan) [inline firebase.json:23]: Will the subscribe page me moving to the marketing site? Can we remove this page?

**Review by khanhnwin — COMMENTED** [review]
A few minor comments

File: firebase.json (hunk 19 above)
Comment (johnpryan) [reply firebase.json:22]: Nevermind, I forgot that this is being deployed to `docs.flutter.dev` now.

File: firebase.json (hunk 20 above)
Comment (khanhnwin) [reply firebase.json:23]: Can `/subscribe` be a redirect to the anchor `https://flutter.dev/#newsletter`?

File: src/cookbook/networking/send-data.md [hunk 21]
```diff
@@ -354,16 +354,16 @@ class _MyAppState extends State<MyApp> {
 
 [ConnectionState]: {{site.api}}/flutter/widgets/ConnectionState-class.html
 [`didChangeDependencies()`]: {{site.api}}/flutter/widgets/State/didChangeDependencies.html
-[Fetch Data]: /docs/cookbook/networking/fetch-data
+[Fetch Data]:/cookbook/networking/fetch-data
```
[the remark this answers is not in the document]
Comment (btobin) [reply src/cookbook/networking/send-data.md]: Apparently I had a space in that search...

File: firebase.json (hunk 20 above)
Comment (btobin) [reply firebase.json]: Anchor is a good idea, but I do agree that these are not necessary anymore. Removed.

File: src/development/tools/sdk/release-notes/release-notes-1.12.13.md (hunk 18 above)
Comment (khanhnwin) [reply src/development/tools/sdk/release-notes/release-notes-1.12.13.md]: Thanks Brian! Looks like `https://github.com/flutter/website/blob/staging-add-to-app-do-not-delete-until-2020/src/development/packages-and-plugins/plugin-api-migration.md` doesn't exist anymore, John pointed me over to `https://github.com/flutter/website/blob/main/src/development/packages-and-plugins/plugin-api-migration.md` which is the correct link (assuming the new director structure without `/docs`). Can we replace for all 3 occurrences? Sorry missed that on the first review. 

Actual live link right now: `https://github.com/flutter/website/blob/main/src/docs/development/packages-and-plugins/plugin-api-migration.md`

**Review by johnpryan — APPROVED** [review]

**Review by khanhnwin — APPROVED** [review]
LGTM!

**Review by johnpryan — APPROVED** [review]

**Pull request merged**